stock_inventory/models/stock_quant.py

```diff
@@ -1,70 +1,89 @@
-from odoo import _, api, fields, models
+from odoo import fields, models
 
 
 class StockQuant(models.Model):
     _inherit = "stock.quant"
 
+    virtual_in_progress_inventory_id = fields.Many2one(
+        comodel_name="stock.inventory",
+        compute="_compute_virtual_in_progress_inventory_id",
+    )
+    stock_inventory_ids = fields.Many2many(
+        comodel_name="stock.inventory",
+        string="Inventory Adjustment",
+        readonly=True,
+    )
     to_do = fields.Boolean(default=True)
 
+    def _compute_virtual_in_progress_inventory_id(self):
+        Inventory = self.env["stock.inventory"]
+        for rec in self:
+            rec.virtual_in_progress_inventory_id = Inventory.search(
```

Comment:
Prefer a search outside the loop and then filtered_domain on locations.

Reply:
Done